README.md

```diff
@@ -2,4 +2,160 @@
+* any check name (eg. `flag-new-keys`) can be placed at any key (`errors`, `infos`, or `warnings`) in the `__preferences` blob.
+* if no `__preferences` are specified, Kombucha runs [the default checks](#default-checks).
+* for a complete [list of checks](#list-of-checks), scroll down a bit.
+
+### Example output
+
+⚠️ TODO!
+
+### Command line parameters
+
+* `--print-errors-only` (`-e`): ask Kombucha to omit `INFO` and `WARNING` output when printing to the console. Useful if a set of tests is producing a lot of output and you’d like to “zero in” on the errors only, without having to edit the configuration file
+* `--record` (`-r`): ask Kombucha to rewrite all the stored snapshots it knows about. Useful for setting a new baseline for tests or if you know a large amount of the API under test has changed
+* `--snapshots-directory` (`-s`): tell Kombucha where to find snapshots for the test run. If not specified, the default is `./__Snapshots/`.
+* `--work-directory` (`-w`): tell Kombucha what directory to use as the “work directory” (storage for live responses) for the test run. If not specified, the default is `./__Work__/`.
+
+### Running on macOS
+
+Kombucha is written in [Swift](https://swift.org). If you have a Swift 5 development environment available, you can run Kombucha natively on macOS. To do so, clone this repository, and then run
+```
+swift build                         # build Kombucha in debug mode
+./build/debug/kombucha sample.json  # run the sample tests that are included with this repo
+```
+To move forward from this point, construct your own JSON configuration file describing your API endpoints and point Kombucha at that file. We suggest storing this configuration file and its snapshots in a git repository separate from this one, so that you can version your usage of Kombucha separately from any versioning of the tool itself. Such an invocation of the tool would look something like:
+```
+kombucha \
+  ../my-tests/kombucha.json \
+  --snapshots-directory ../my-tests/__Snapshots__ \
+  --work-directory ../my-tests/__Work__
+```
+
+### Running via Docker
+
+⚠️ TODO!
+
+## List of checks
+
+### The `structure` check
+
+We call Kombucha’s main snapshotting behavior “the structure check”. With both the snapshot and the work (live response) JSON in memory, Kombucha will flag any JSON keys that appear in the snapshot structure, but not in the work structure. In addition, it will flag any JSON keys where the type of the value in the work structure differs from the type in the snapshot structure (eg. if the key `"foo"` used to have the value `true`, but now it has the value `99.0` or `["hello"]`).
+
+Kombucha performs this check recursively into nested JSON objects as deep as they go, and recursively into the *first element only* of nested JSON arrays, as deep as they go. The lack of coverage for “all” of an array is an issue, but we’re not sure how to solve or improve this yet.
```

Comment:
Maybe say that this check + array-consistency is a possible solution? Since if it passes on the first element and the array are heterogeneous, you have a consistent structure :)